controllers/imagepolicy_controller.go

```diff
@@ -284,3 +306,58 @@ func (r *ImagePolicyReconciler) patchStatus(ctx context.Context, req ctrl.Reques
 
 	return r.Status().Patch(ctx, &res, patch)
 }
+
+func (r *ImagePolicyReconciler) hasAccessToRepository(ctx context.Context, policy ctrl.Request, repo meta.NamespacedObjectReference, acl *imagev1.AccessFrom) (bool, error) {
+	// grant access if the policy is in the same namespace as the repository
+	if repo.Namespace == "" || policy.Namespace == repo.Namespace {
+		return true, nil
+	}
+
+	// deny access if the repository has no ACL defined
+	if acl == nil {
+		return false, fmt.Errorf("ImageRepository '%s/%s' can't be accessed due to missing access list",
+			repo.Namespace, repo.Name)
+	}
+
+	// grant access if the repository ACL has no namespace selectors
+	if acl != nil && len(acl.NamespaceSelectors) == 0 {
```

Comment:
Surely no selectors (rather than an empty selector) means "allow none"?

Reply:
What was the resolution?

Reply:
`if acl != nil && acl.NamespaceSelectors == nil`